cms/models.py

```diff
@@ -191,7 +191,35 @@ def get_child_by_readable_id(self, readable_id):
         return self.get_children().get(coursepage__course__readable_id=readable_id)
 
 
-class ProductPage(Page):
+class FactultyPage(Page):
+    """
+    Abstract detail page for faculty members
+    """
+
+    faculty_section_title = models.CharField(
+        max_length=255,
+        null=True,
+        blank=True,
+        default="Meet your instructors",
+        help_text="The title text to display in the faculty cards section of the product page.",
+    )
+
+    faculty_members = StreamField(
+        [("faculty_member", FacultyBlock())],
+        null=True,
+        help_text="The faculty members to display on this page",
+    )
+
+    content_panels = Page.content_panels + [
+        FieldPanel("faculty_section_title"),
+        StreamFieldPanel("faculty_members"),
+    ]
+
+    class Meta:
+        abstract = True
+
+
+class ProductPage(FactultyPage):
```

Comment:
I'm not sure about this pattern. If we keep implementing possibly-reusable sections of pages like this, then we'll have page classes with lots of different parent classes. I don't think that's sustainable. I think we should get rid of the `FactultyPage` class and move its fields into `ProductPage`. If another page ends up having a faculty section, it would just need to define those same 2 fields, which I think is completely reasonable